File: leetcode/medium/1605. Find Valid Matrix Given Row and Column Sums/solution.cc
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        int r = rowSum.size(), c = colSum.size();
        vector<vector<int>> restored(r, vector<int>(c, 0));
        for (int y = 0; y < r; y += 1) {
            for (int x = 0; x < c; x += 1) {
                restored[y][x] = min(rowSum[y], colSum[x]);
                rowSum[y] -= restored[y][x];
                colSum[x] -= restored[y][x];
            }
        }
        return restored;
    }
};

int main() {
    return 0;
}

File: leetcode/medium/1605. Find Valid Matrix Given Row and Column Sums/solution.js
// https://leetcode.com/problems/find-valid-matrix-given-row-and-column-sums/
/**
 * @param {number[]} rowSum
 * @param {number[]} colSum
 * @return {number[][]}
 */
function restoreMatrix(rowSum, colSum) {
    const restored = Array.from(Array(rowSum.length), () => new Array(colSum.length));
    for (let y = 0; y < rowSum.length; y += 1) {
        for (let x = 0; x < colSum.length; x += 1) {
            const min = Math.min(rowSum[y], colSum[x]);
            rowSum[y] -= min;
            colSum[x] -= min;
            restored[y][x] = min;
        }
    }
    console.log(restored);
    return restored;
};

restoreMatrix([14,9],[6,9,8])
restoreMatrix([0], [0])
restoreMatrix([1,0], [1])
restoreMatrix([1], [1, 0])
